modules/imaging_uploader/php/imaging_uploader.class.inc

```diff
@@ -326,7 +326,32 @@ class Imaging_Uploader extends \NDB_Menu_Filter_Form
                                 " created in the database. Please refresh this".
                                 " page to return to the Imaging Uploader module";
                 }
-            }
+                ///////////////////////////////////////////////////////////////////////
+                /////////////////// make sure the candidate is active /////////////////
+                ///////////////////////////////////////////////////////////////////////
+                else {
+                    $active = $db->pselectOne(
+                        "SELECT COUNT(*) FROM session s
+                        JOIN candidate c ON (c.candid=s.candid)
+                        WHERE s.Visit_label = :vlabel
+                        AND c.pscid = :pid AND c.candid = :cid AND c.active = 'Y'",
+                        [
+                         ':cid'    => $candid,
+                         ':vlabel' => $visit_label,
+                         ':pid'    => $pscid,
+                        ]
+                    );
```

Comment:
Should also look for active session (s.active).

Instead of adding more query, could modify existing query to get individual attributes instead of count and then build proper message to display. (using left join)

If number of row = 0, candidate nonexistent
visitlabel is null, no visit 
c.active  = 'N', candidate not active
s.active = 'N' session not active

Reply:
👍 for Melanie's solution. Giving more details as to why the error occurred makes sense.

Reply:
also agree

Reply:
Great, that would combine the query @victoriafoing made with the previous one starting on line 308 for an elegant solution.